code/datums/action.dm

```diff
@@ -515,26 +513,28 @@
 		return spell.can_cast(owner)
 	return FALSE
 
-/datum/action/spell_action/UpdateButtonIcon()
-	if(button && !(. = ..()))
-		var/obj/effect/proc_holder/spell/S = target
-		if(!istype(S))
-			return
-		var/progress = S.get_availability_percentage()
-		var/col_val_high = 72 * progress + 128
-		var/col_val_low = 200 * progress
-		button.maptext = "<div style=\"font-size:6pt;color:[recharge_text_color];font:'Small Fonts';text-align:center;\" valign=\"bottom\">[round_down(progress * 100)]%</div>"
-		button.color = rgb(col_val_high, col_val_low, col_val_low, col_val_high)
-	else
-		button.maptext = null
+/datum/action/spell_action/apply_unavailable_effect()
+	var/obj/effect/proc_holder/spell/S = target
+	if(!istype(S))
+		return ..()
+	var/progress = S.get_availability_percentage()
+	if(progress == 1)
+		return ..() // This means that the spell is charged but unavailable due to something else
+
+	var/alpha = 220 - 140 * progress
+
+	var/image/img = image('icons/mob/screen_white.dmi', icon_state = "template")
+	img.alpha = alpha
+	img.appearance_flags = RESET_COLOR | RESET_ALPHA
+	img.color = "#000000"
+	img.plane = FLOAT_PLANE + 1
+	button.add_overlay(img)
+	// Make a holder for the charge text
```

Comment:
It's clear that you're making the holder for the text. It's not clear why you need the holder at all, and why can't you slap that text onto `img`.
Clarify the rationale in the comment (i.e. that the alpha shenanigans make the text unreadable, or something)

Reply:
Fair point. Will add it when the inevitable third PR comes around to change spells :P